tests/flow/test_entity_update.py

```diff
@@ -113,9 +113,30 @@ def test11_property_map_from_parameter(self):
         self.env.assertEqual(result.result_set, expected_result)
 
     # Fail update an entity property when left hand side is not alias
-    def test12_fail_update_property_of_non_alias_enetity(self):
+    def test12_fail_update_property_of_non_alias_entity(self):
         try:
             graph.query("MATCH P=() SET nodes(P).prop = 1 RETURN nodes(P)")
             self.env.assertTrue(False)
         except ResponseError as e:
             self.env.assertContains("RedisGraph does not currently support non-alias references on the left-hand side of SET expressions", str(e))
+
+    # Fail when a property is a complex type nested within an array type
+    def test13_invalid_complex_type_in_array(self):
+        # Skip this test if running under Valgrind, as it causes a memory leak.
+        if Env().envRunner.debugger is not None:
+            Env().skip()
```

Comment:
Are we leaking here?

Reply:
We had been; I rewrote the validation such that it no longer leaks.

As with many other run-time validations, it is still possible to cause leaks when jumping to the exception handler. For example, this query would leak:
```
WITH {k: 'v'} AS map MATCH (a) SET a.v = [map]
```
(In this case, `map` will leak due to its allocation at https://github.com/RedisGraph/RedisGraph/blob/master/src/execution_plan/ops/op_update.c#L75)

I don't believe these leaks can be entirely plugged without a change like SIValue reference counting.